app/build.gradle

```diff
@@ -0,0 +1,162 @@
+plugins {
+    id 'com.android.application'
+}
+
+android {
+    namespace 'juloo.keyboard2'
+    compileSdk 33
+
+    defaultConfig {
+        applicationId "juloo.keyboard2"
+        minSdk 4
+        targetSdkVersion 33
+        versionCode 36
+        versionName "1.24.0"
+    }
+    signingConfigs {
+        // Debug builds will always be signed. If no environment variables are set, a default
+        // keystore will be initialized by the task initDebugKeystore and used. This keystore
+        // can be uploaded to GitHub secrets by following CONTRIBUTING.md#Using the local debug.keystore on the Github CI actions
+        // in order to always receive correctly signed debug APKs from the CI.
+        debug {
+            storeFile (System.env.DEBUG_KEYSTORE ? file(System.env.DEBUG_KEYSTORE) : file("../debug.keystore"))
+            storePassword (System.env.DEBUG_KEYSTORE_PASSWORD ? "$System.env.DEBUG_KEYSTORE_PASSWORD" : "debug0")
+            keyAlias (System.env.DEBUG_KEY_ALIAS ? "$System.env.DEBUG_KEY_ALIAS" : "debug")
+            keyPassword (System.env.DEBUG_KEY_PASSWORD ? "$System.env.DEBUG_KEY_PASSWORD" : "debug0")
+        }
+        // Release builds will only be signed with a dedicated key if specified. This key has to be
+        // created manually. To be used with GitHub CI, key and environment variables have to be
+        // added as secrets. See file .github/workflows/make-apk.yml for further reference.
+        release {
+            if (System.env.RELEASE_KEYSTORE) {
+                storeFile file(System.env.RELEASE_KEYSTORE)
+                storePassword "$System.env.RELEASE_KEYSTORE_PASSWORD"
+                keyAlias "$System.env.RELEASE_KEY_ALIAS"
+                keyPassword "$System.env.RELEASE_KEY_PASSWORD"
+            }
+        }
+    }
+    buildTypes {
+        release {
+            minifyEnabled true
+            shrinkResources true
+            debuggable false
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+            if (System.env.RELEASE_KEYSTORE) {
+                signingConfig signingConfigs.release
+            }
+        }
+        debug {
+            minifyEnabled false
+            shrinkResources false
+            debuggable true
+            applicationIdSuffix ".debug"
+            signingConfig signingConfigs.debug
+        }
+    }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+}
+
+dependencies {
+
+}
+
+tasks.named("preBuild") {
+    dependsOn += "buildKeyboardAssets"
+}
+
+tasks.register('buildKeyboardAssets') {
+    dependsOn 'copyKeyboardFont'
+    dependsOn 'buildKeyboardLayoutXml'
+}
+
+tasks.register('buildKeyboardFont') {
+    println "Executing fontforge to build custom font\n"
```

Comment:
This is not caching the built font in the source tree. This was important as the appropriate version of fontforge can't be installed in every distros (for example in f-droid build scripts).

Reply:
I made this task optional so it can be invoked manually, but just if needed. The built font file is included now.